client/signup/steps/domains/index.jsx

```diff
@@ -264,17 +281,62 @@ class DomainsStep extends Component {
 		this.props.recordUseYourDomainButtonClick( this.getAnalyticsSection() );
 	};
 
+	handleDomainToDomainCart = () => {
+		const { step } = this.props;
+
+		const { suggestion } = step;
+		const isPurchasingItem = suggestion && Boolean( suggestion.product_slug );
+		const siteUrl =
+			suggestion &&
+			( isPurchasingItem
+				? suggestion.domain_name
+				: suggestion.domain_name.replace( '.wordpress.com', '' ) );
+
+		if ( hasDomainInCart( this.props.cart, suggestion.domain_name ) ) {
+			this.removeDomain( suggestion );
+		} else {
+			this.addDomain( suggestion );
+			this.props.setDesignType( this.getDesignType() );
+			// Start the username suggestion process.
+			siteUrl && this.props.fetchUsernameSuggestion( siteUrl.split( '.' )[ 0 ] );
+		}
+	};
+
+	shouldUseMultipleDomainsInCart = () => {
+		const { step, flowName } = this.props;
+		if ( ! step ) {
+			return;
+		}
+		const { suggestion } = step;
+
+		const enabledFlows = [ 'onboarding', 'onboarding-pm' ];
```

Comment:
I was thinking we'd just introduce this to the 'onboarding' flow to start. Any reason we need to include 'onboarding-pm'?

Reply:
Sure, I will keep only `onboarding` flow